pkg/cluster/fixinfraid.go

```diff
@@ -0,0 +1,29 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+)
+
+func (m *manager) fixInfraID(ctx context.Context) error {
+	infraID := m.doc.OpenShiftCluster.Properties.InfraID
+	if infraID != "" {
+		return nil
+	}
+	infraID = "aro"
+
+	var err error
+	m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
+		doc.OpenShiftCluster.Properties.InfraID = infraID
+		return nil
+	})
+	if err != nil {
+		m.log.Error(err)
```

Comment:
Do we still want to proceed and not return an error if for some reason the infraID patching fails ? i.e. we we don't return error error ?

Reply:
Good catch. Why not `return err` here?

Reply:
Because in adminUpdate we dont want to fail if we don't need to. In reality this can flake on database flake and this would cause a failure in adminUpdate. Other steps follows the same path.

Reply:
@mjudeikis I would rather fail explicitly here rather than mask the failure. If we reach `PatchWithLease` in this function, it means that the document is missing infra id and we still need to update it. Without having all documents updated, we will not be able to get rid of having this everywhere:

```go
infraID := m.doc.OpenShiftCluster.Properties.InfraID
if infraID == "" {
    infraID = "aro"
}
```

Reply:
If we decide take the path of not returning error here (again: I think we should not mask error here), then still need to update the function: we always `return nil`. I guess we can get rid of the return value.

Reply:
@mjudeikis That's true but other tests fail in similar situation like populateCreateAt; hence my original question. I would tend to agreed with @m1kola, especially since we want to get rid of this error

Reply:
My take on this is, What can cause a failure here?
It might be networking flake - in this case should we fail put or patch? I think no as if we didnt got this cluster this time, we will get next time. And failure is cognitive load for SRE.
Any other more serious failure means database is corrupted and it will fail way before this step is reached. 
But ok, I dont see this step ever failing so happy to change :)

Reply:
> It might be networking flake - in this case should we fail put or patch? I think no as if we didnt got this cluster this time, we will get next time. And failure is cognitive load for SRE.

Agree, it is cognitive load for SRE, but the sooner we retry that flake and update the DB the better: it is greatly reduces the chances of introducing similar regression which results in even higher cognitive load.

Thanks for making this change.